src/screens/DiscoveryScreenV2/elements/Search.js

```diff
@@ -214,9 +215,27 @@ const DiscoverySearch = ({
   );
 };
 
+DiscoverySearch.propTypes = {
+  setDiscoveryLoadingData: PropTypes.func,
+  searchText: PropTypes.string,
+  setSearchText: PropTypes.func,
+  placeholderText: PropTypes.string,
+  setIsFocus: PropTypes.func,
+  setIsFirstTimeOpen: PropTypes.func,
+  fetchDiscoveryData: PropTypes.func,
+  fetchData: PropTypes.func,
+  onCancelToken: PropTypes.func,
+  hideBackIcon: PropTypes.bool,
+  autoFocu: PropTypes.bool
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

You have added PropTypes for the new props and existing ones. This is a good practice as it helps in type checking the props and makes the code more robust and less prone to bugs due to incorrect data types. However, there seems to be a typo in the `autoFocus` propType declaration. It should be `autoFocus`, not `autoFocu`.
```diff
  onCancelToken: PropTypes.func,
  hideBackIcon: PropTypes.bool,
- autoFocu: PropTypes.bool
+ autoFocus: PropTypes.bool
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->